sw/host/cryptotest/testvectors/parsers/wycheproof_ecdsa_parser.py

```diff
@@ -46,29 +59,36 @@ def parse_test_vectors(raw_data):
         for test in group["tests"]:
             logging.debug(f"Parsing tcId {test['tcId']}")
             test_vec = {
+                "vendor": "wycheproof",
+                "test_case_id": test['tcId'],
                 "algorithm": "ecdsa",
                 "operation": "verify",
                 "curve": EC_NAME_MAPPING[key["curve"]],
                 "hash_alg": group["sha"].lower(),
                 "message": list(bytes.fromhex(test["msg"])),
-                "qx": int(key["wx"], 16),
-                "qy": int(key["wy"], 16),
+                # These values are encoded as a string so the test harness can
+                # extract the entire value (serde_json likes to turn large
+                # integers into limited-precision floating point values)
+                "qx": key["wx"],
+                "qy": key["wy"],
             }
 
             # Parse r and s from DER-encoded signature
             # If the DER sequence fails to parse, then it may be malformed as
             # part of the test case. Cryptolib does not accept DER-encoded
             # input, so ignore this test case.
             try:
-                signature_seq = _parse_der_signature(test["sig"])
+                signature_seq = _parse_der_signature(test["sig"], key["curve"])
             except ValueError:
                 logging.info(f"Skipped tcId {test['tcId']}: ValueError while parsing DER sequence.")
                 continue
             except IndexError:
                 logging.info(f"Skipped tcId {test['tcId']}: IndexError while parsing DER sequence.")
                 continue
-
-            test_vec["r"], test_vec["s"] = signature_seq
+            
+            # Same as above: encode these values as hex strings so serde_json does
+            # not truncate the values (the [2:] is to remove the '0x').
```

Comment:
Another small comment nit: Avoid saying "Same as above" since the code above could be changed in the future.

```python
# Encode r and s as hex strings since downstream test vector
# consumers may incorrectly handle large values.
# [2:] removes the "0x" prefix.
```